master/buildbot/steps/artifact.py

```diff
@@ -183,9 +187,15 @@ def start(self):
 
         if self.artifactBuildrequest:
             self.step_status.setText(["Artifact has been already generated."])
-            self.artifactPath = "%s_%s_%s" % (self.build.builder.config.builddir,
-                                              self.artifactBuildrequest['brid'],
-                                              FormatDatetime(self.artifactBuildrequest['submitted_at']))
+
+            if self.artifactBuildrequest["submitted_at"] > ARTIFACT_LOCATION_CHANGE_DATE:
+                self.artifactPath = "%s/%s_%s" % (self.build.builder.config.builddir,
```

Comment:
@campos-ddc when we deploy changes we will stop katana (which stops all running builds) and during startup we do CheckArtifactExists and then run a rebuild if the artifact is not on the server. we will be just rebuilding until it catch up with the new artifact structure, if we deploy changes when katana is at lower load then I think we should be ok